WordPress/gutenberg
# [Data Views] User patterns: Use excerpt as description

## What?

Enable descriptions for user created patterns by using the excerpt field.
Partial for https://github.com/WordPress/gutenberg/issues/55244

## Why?

User created patterns did not have an option to add descriptions. Text only descriptions are important for users who are unable to see or interpret the content in the pattern preview.

## How?

This PR:
- Enables excerpts on the `wp_block` post type (This PHP change will eventually need a backport to `create_initial_post_types()` in wp-includes / post.php).
- On the Site Editor Pattern pages, use the excerpt as the description for user created patterns by:
   - Fetching the raw excerpt in `convertPatternPostToItem` and assigning it to the description.
   - Updating the condition that prints the description, to also print it on user created patterns.

## Todo:
The PR does not populate the excerpt field with the description when a theme or plugin pattern is duplicated into a user created pattern.

## Testing Instructions
The excerpt field can be tested both with a classic theme and block theme, when creating a new pattern or editing an existing pattern. The field should show in the "Pattern" sidebar panel, below the categories.

The Site Editor Patterns page can only be tested with a block theme:
Activate a block theme.
Create a new pattern. (It doesn't matter if the pattern is synced or not)
Add an excerpt to the pattern, save.
Go to Editor > Patterns, select "My patterns".
If you have not already done so, activate your screen reader.
Navigate to the pattern that you just created: When you hear that the title of your pattern is announced, it should be followed by the excerpt.

You can also confirm that the  excerpt text is output in the markup. It should be printed in a div inside another div with the class `dataviews-view-grid__media`

## Screenshots or screencast 

<img width="279" alt="pattern-excerpt" src="https://github.com/WordPress/gutenberg/assets/7422055/4fff0adb-648a-4f0f-970e-2e80915df6c7">
